backend/utils/db.py

```diff
@@ -1,51 +1,97 @@
 from fastapi import status
 from fastapi.exceptions import HTTPException
 
-PLUGIN_KEY = "chat.zuri.chat"
-BASE_URL = "https://staging.api.zuri.chat"
-
 
 class DataStorage:
-    """Serves as a layer for communication for plugin data and server
+    """Serves as a layer of abstraction for communication between zc_messaging
+    data and the database on zc_core.
+
+    It uses API endpoints from zc_core to perform CRUD operations on zc_messaging
+    collection data.
 
-    Provides a layer of abstraction for communication between plugin
-    data and the database on zc_core
+    Attributes:
+        write_api str): Zc_core API endpoint for writing (POST) and updating (PUT) data.
+        read_api (str): Zc_core API endpoint for reading data.
+        delete_api (str): Zc_core API endpoint for deleting data.
+        get_members_api (str): Zc_core API endpoint for getting members of an organization.
+        organization_id (str): The organization id where the operations are to be performed.
+        plugin_id (str): The zc_messaging plugin id in the plugins marketplace.
 
     """
 
     def __init__(self, organization_id: str) -> None:
-        self.write_api = f"{BASE_URL}/data/write"
-        self.delete_api = f"{BASE_URL}/data/delete"
-        self.read_query_api = f"{BASE_URL}/data/read"
-        self.get_member_api = f"{BASE_URL}/organizations/{organization_id}/members/"
+        """Initializes the data storage instance with zc_messaging plugin id.
+
+        A request is sent to the plugins marketplace API endpoint on zc_core to get
+        the plugin id.
+
+        Args:
+            organization_id: The organization id where the operations are to be performed.
+
+        Raises:
+            HTTPException: {"detail": "Request Timeout"}
+        """
+        self.write_api = f"{settings.BASE_URL}/data/write"
+        self.read_api = f"{settings.BASE_URL}/data/read"
+        self.delete_api = f"{settings.BASE_URL}/data/delete"
+        self.get_members_api = (
+            f"{settings.BASE_URL}/organizations/{organization_id}/members/"
+        )
         self.organization_id = organization_id
 
         try:
-            response = requests.get(url=f"{BASE_URL}/marketplace/plugins")
+            response = requests.get(url=f"{settings.BASE_URL}/marketplace/plugins")
             plugins = response.json().get("data").get("plugins")
             plugin = next(
-                item for item in plugins if PLUGIN_KEY in item["template_url"]
+                item for item in plugins if settings.PLUGIN_KEY in item["template_url"]
             )
-            self.plugin_id = plugin["id"] if plugin else None
+            self.plugin_id = plugin.get("id")
         except requests.exceptions.RequestException as exception:
-            print(exception)
             raise HTTPException(
                 status_code=status.HTTP_408_REQUEST_TIMEOUT, detail="Request Timeout"
-            )
+            ) from exception
```

Comment:
@man-of-mind, we need to change this part.
When a Request exception occurs, the reason could be, not only a timeout but also an HTTP error, a Connection error, etc. 
Do we want to handle each type of RequestException explicitly?

Reply:
If there are possibilities of other exceptions, it is better we handle them differently

Reply:
Okay boss, should i do this or u will add it to this PR?
@zxenonx

Reply:
I'm gonna do it. @man-of-mind

Reply:
Alright boss @zxenonx

Reply:
I've made the changes. Please check.